client/src/api/helpers.js

```diff
@@ -1,3 +1,7 @@
+import request from './request';
+
 const extractData = res => res.data;
 
-export { extractData };
+const fetchAll = (url) => request.get(url).then(extractData);
+
+export { extractData, fetchAll };
```

Comment:
so a fetchOne would be what, this?
```js
const fetchOne = (url) => request.get(url).then(extractData);
```
The point I'm trying to make is that naming this fetch all doesn't really make sense. The best name for this in my opinion would be `getRequest` or `getByUrl` but that doesn't really help. I guess you wanted to extract this so you can have the `then(extractData)` be consistent. If that is the case I have to point out that post requests would need their own function. Delete as well. Get with params also. Post and delete that dont return something you can extractData as well, etc. That just makes everything a bit more complicated than it needs to be. 

Api functions are so simple I don't see the reason to generalise them, I think it's better to have `request.get(url).then(format)` inside of lets say recipes because when I work on something new and I need to do a get request I want to open /api/recipes.js and find the fetchAll saying its a get request going to an url defined in the same file. Everything being in the same place is a good thing here because of the context

Reply:
To conclude a very long point, I don't think having a fetchAll function is necessary nor productive. In my opinion it's just abstraction for the sake of abstraction, I don't see the genuine use/help it provides